### DISCUSSION TITLE
How to replace ObjectJointRevolute2D with GenericJoint

### DISCUSSION THREAD

**Discussion by ManuelZ** [jgerstmayr/EXUDYN#45] [Q&A on usage of Exudyn] [answered]
While trying to get the joint reaction forces of this mechanism (which has 0 DOF and is being solved with SolveStatic):

![image](https://user-images.githubusercontent.com/115771/235573554-2c4d17b7-968f-4962-a80b-951a1bb5e7bb.png)

...my first thought was to get the information from the joints, but [you said](https://github.com/jgerstmayr/EXUDYN/issues/13#issuecomment-1054416543) that "CObjectJointRevolute2D has currently no output variables" (although in that case, the question was about positions). 

A `GenericJoint` joint seems to provide a `ForceLocal` output variable, so I may just replace my `RevoluteJoint2D` with it, however I'm having doubts about if I should do it like this:

```
GenericJoint(
    constrainedAxes = [1, 1, 0, 0, 0, 0],
    ...
)
```

Would this be correct?

I'm asking, in part, because you also said "The rotation is not available in JointRevolute2D, as the 2D revolute joint can also be attached to position-based markers, which have no rotation." But I can see rotation being available in `GenericJoint`. So a `JointRevolute2D` may be more involved than just defining the constrained freedoms?

**Comment by jgerstmayr** [comment, marked as answer]
This was still open and not answered. But it is generally correct, to use a GenericJoint in this case. However, it leads to more constraint equations, as it always produces 6 equations.
